### PR TITLE
New APIs for footer/header view manipulation

### PR DESCRIPTION
**Pull request by zplesac**
Defined new methods for footer/header view manipulation.

### CODE REVIEW

File: mjolnirrecyclerview/src/main/java/co/infinum/mjolnirrecyclerview/MjolnirRecyclerAdapter.java [hunk 1]
```diff
@@ -292,49 +292,159 @@ private int calculateIndex(int index) {
...
 
     /**
+     * Add a footer view to this adapter. If footer already exists, it will be replaced depending on the {@param shouldReplace} value.
+     *
+     * @param footerViewId  layout view id.
+     * @param shouldReplace should we replace footer if it already exists
+     * @return true if footer was added/replaced, false otherwise.
+     */
+    public boolean addFooter(@LayoutRes int footerViewId, boolean shouldReplace) {
+        if (!hasFooter() || hasFooter() && shouldReplace) {
```
Comment (ikocijan) [inline mjolnirrecyclerview/src/main/java/co/infinum/mjolnirrecyclerview/MjolnirRecyclerAdapter.java]: I would reverse this logic to make it a bit cleaner. 

```java
if (shouldReplace || !hasFooter()) {
}
```

File: mjolnirrecyclerview/src/main/java/co/infinum/mjolnirrecyclerview/MjolnirRecyclerAdapter.java [hunk 2]
```diff
@@ -292,49 +292,159 @@ private int calculateIndex(int index) {
...
 
     /**
+     * Add a footer view to this adapter. If footer already exists, it will be replaced depending on the {@param shouldReplace} value.
+     *
+     * @param footerView    layout view
+     * @param shouldReplace should we replace footer if it already exists
+     * @return true if footer was added/replaced, false otherwise.
+     */
+    public boolean addFooter(View footerView, boolean shouldReplace) {
+        if (!hasFooter() || hasFooter() && shouldReplace) {
```
Comment (ikocijan) [inline mjolnirrecyclerview/src/main/java/co/infinum/mjolnirrecyclerview/MjolnirRecyclerAdapter.java]: The same as the comment for  addFooter(@LayoutRes int footerViewId, boolean shouldReplace) method

File: mjolnirrecyclerview/src/main/java/co/infinum/mjolnirrecyclerview/MjolnirRecyclerAdapter.java [hunk 3]
```diff
@@ -292,49 +292,159 @@ private int calculateIndex(int index) {
...
 
     /**
+     * Add a header view to this adapter. If header already exists, it will be replaced depending on the {@param shouldReplace} value.
+     *
+     * @param headerViewId  layout view id.
+     * @param shouldReplace should we replace header if it already exists
+     * @return true if header was added/replaced, false otherwise.
+     */
+    public boolean addHeader(@LayoutRes int headerViewId, boolean shouldReplace) {
+        if (!hasHeader() || hasHeader() && shouldReplace) {
```
Comment (ikocijan) [inline mjolnirrecyclerview/src/main/java/co/infinum/mjolnirrecyclerview/MjolnirRecyclerAdapter.java]: Same here

File: mjolnirrecyclerview/src/main/java/co/infinum/mjolnirrecyclerview/MjolnirRecyclerAdapter.java [hunk 4]
```diff
@@ -292,49 +292,159 @@ private int calculateIndex(int index) {
...
 
     /**
+     * Add a header view to this adapter. If header already exists, it will be replaced depending on the {@param shouldReplace} value.
+     *
+     * @param headerView    layout view
+     * @param shouldReplace should we replace header if it already exists
+     * @return true if header was added/replaced, false otherwise.
+     */
+    public boolean addHeader(View headerView, boolean shouldReplace) {
+        if (!hasHeader() || hasHeader() && shouldReplace) {
```
Comment (ikocijan) [inline mjolnirrecyclerview/src/main/java/co/infinum/mjolnirrecyclerview/MjolnirRecyclerAdapter.java]: And here

File: mjolnirrecyclerview/src/main/java/co/infinum/mjolnirrecyclerview/MjolnirRecyclerAdapter.java (hunk 3 above)
Comment (zplesac) [inline mjolnirrecyclerview/src/main/java/co/infinum/mjolnirrecyclerview/MjolnirRecyclerAdapter.java]: Same comment as above

File: mjolnirrecyclerview/src/main/java/co/infinum/mjolnirrecyclerview/MjolnirRecyclerAdapter.java (hunk 4 above)
Comment (zplesac) [inline mjolnirrecyclerview/src/main/java/co/infinum/mjolnirrecyclerview/MjolnirRecyclerAdapter.java]: Same comment as above

File: mjolnirrecyclerview/src/main/java/co/infinum/mjolnirrecyclerview/MjolnirRecyclerAdapter.java (hunk 3 above)
Comment (ikocijan) [inline mjolnirrecyclerview/src/main/java/co/infinum/mjolnirrecyclerview/MjolnirRecyclerAdapter.java]: Same comment as above

File: mjolnirrecyclerview/src/main/java/co/infinum/mjolnirrecyclerview/MjolnirRecyclerAdapter.java (hunk 4 above)
Comment (ikocijan) [inline mjolnirrecyclerview/src/main/java/co/infinum/mjolnirrecyclerview/MjolnirRecyclerAdapter.java]: Same comment as above

File: mjolnirrecyclerview/src/main/java/co/infinum/mjolnirrecyclerview/MjolnirRecyclerAdapter.java (hunk 1 above)
Comment (zplesac) [inline mjolnirrecyclerview/src/main/java/co/infinum/mjolnirrecyclerview/MjolnirRecyclerAdapter.java]: We've discussed about this on Slack, you are right.

**Pull request merged**